File: 3sem/4lab/Banks/BankAccounts/Factories/DebitBankAccountFactory.cs
using Banks.Clocks;
using Banks.Entities;

namespace Banks.BankAccounts.Factories;

public class DebitBankAccountFactory : IBankAccountFactory
{
    public IBankAccount Create(Bank bank, Client client, IClock clock)
    {
        return new DebitBankAccount(bank, client, clock);
    }
}

File: 4sem/1lab/src/main/java/BanksRemake/Domain/BankAccounts/Factories/DebitBankAccountFactory.java
package BanksRemake.Domain.BankAccounts.Factories;

import BanksRemake.Domain.BankAccounts.BankAccount;
import BanksRemake.Domain.BankAccounts.DebitBankAccount;
import BanksRemake.Domain.Clocks.Clock;
import BanksRemake.Domain.Entities.Bank;
import BanksRemake.Domain.Entities.Client;

/**
 * Factory creating debit bank accounts.
 */
public class DebitBankAccountFactory implements BankAccountFactory {
    @Override
    public BankAccount create(Bank bank, Client client, Clock clock) {
        return new DebitBankAccount(bank, client, clock);
    }
}
